python/PyQt6/core/auto_generated/providers/sensorthings/qgssensorthingsutils.sip.in

```diff
@@ -61,6 +61,25 @@ Returns ``True`` if the specified entity ``type`` can have geometry attached.
 %Docstring
 Returns a filter string which restricts results to those matching the specified
 ``entityType`` and ``wkbType``.
+%End
```

Comment:
I'm quite surprised to see changes in Python bindings. Why is the SensorThings provider in src/core/providers/ and not in src/providers ?

Reply:
There's a couple of reasons for this trend of exposing utility functions like these:

1. By exposing these utility functions to python we make it easy for plugins to use them where needed 
2. It makes it easier to unit test

Regarding the actual location of the data provider in core (as opposed to SRC/providers/), this is also an ongoing trend. There's really no compelling reason why (inbuilt) providers should be implemented in a plugin style anymore. We used to do this when there wasn't a clean seperation between core and gui components in plugins, but that's all fixed now. And there's compelling reasons to avoid splitting providers into their own libraries:

- allows easier code reuse between providers
- slim builds can still be made via cmake configuration
- faster startup times due to avoiding dynamic loading of providers

For this reason i'm actually consider a grant request for moving more providers into src/core, specifically the ones which don't have any external dependencies (eg ArcGIS rest, wms, delimited text,...)

Reply:
ok makes sense

> 1. By exposing these utility functions to python we make it easy for plugins to use them where needed

are the new utility methods really useful for plugins? They seem to be internal implementation details, don't they?

Reply:
> are the new utility methods really useful for plugins?  They seem to be internal implementation details, don't they?

I can't speak for the SensorThings classes, but I've definitely used the equivalent QgsArcGisRestUtils classes extensively in custom scripts. :shrug:

Reply:
ok, I trust your experience on this (I tend to be super conservative on what I expose in public APIs)